tests/forte/data/data_store_test.py

```diff
@@ -296,21 +305,58 @@ def test_get_type_info(self):
         self.assertTrue("invalid.Type" not in DataStore._type_attributes)
 
         # test get existing type
-        doc_attr_dict = empty_data_store._get_type_info("ft.onto.base_ontology.Document")
+        doc_attr_dict = empty_data_store._get_type_info(
+            "ft.onto.base_ontology.Document"
+        )
         self.assertEqual(len(DataStore._type_attributes), 2)
-        self.assertEqual(doc_attr_dict, DataStore._type_attributes["ft.onto.base_ontology.Document"])
+        self.assertEqual(
+            doc_attr_dict,
+            DataStore._type_attributes["ft.onto.base_ontology.Document"],
+        )
 
         # test get type info with ontology file input
         with self.assertRaisesRegex(
-            RuntimeError, "DataStore is initialized with no existing types. Setting"
-                "dynamically_add_type to False without providing onto_file_path"
-                "will lead to no usable type in DataStore."
+            RuntimeError,
+            "DataStore is initialized with no existing types. Setting"
+            "dynamically_add_type to False without providing onto_file_path"
+            "will lead to no usable type in DataStore.",
         ):
             DataStore(dynamically_add_type=False)
 
         DataStore._type_attributes = self.reference_type_attributes
         # TODO: need more tests for ontology file input
 
+    def test_entry_methods(self):
+        sent_type = "ft.onto.base_ontology.Sentence"
+        doc_type = "ft.onto.base_ontology.Document"
+        ann_type = "forte.data.ontology.top.Annotation"
+        sent_list = [
+            entry
+            for entry in list(self.data_store._DataStore__entry_dict.values())
+            if entry[3] == "ft.onto.base_ontology.Sentence"
+        ]
+        doc_list = [
+            entry
+            for entry in list(self.data_store._DataStore__entry_dict.values())
+            if entry[3] == "ft.onto.base_ontology.Document"
+        ]
+        ann_list = list(self.data_store._DataStore__entry_dict.values())
```

Comment:
This might not be correct if there are `Link`s or `Group`s in `DataStore`. For example, the format of  test entry with `tid=10123` looks like a `Group` but will also be included in `ann_list`

Reply:
Is there any way to check `Link`, I found `Link` type entry in `self.__elements` is similar to a regular entry.

Reply:
According to the docstring in `DataStore.__init__`, a `Link` entry has the following format: `[<parent_tid>, <child_tid>, <tid>, <type_name>, <attr_1>, <attr_2>, ..., <attr_n>, index_id]`. So I guess you can still use `<type_name>` to check if it's a link.

Reply:
I think the rest of the PR is fine. What's the current status of this comment?

@mylibrar  feel free to resolve the comments if you are fine with them, and you can approve it after your review too.